src/react/components/Gallery.tsx

```diff
@@ -0,0 +1,55 @@
+import { TFile } from "obsidian";
+
+import Masonry from "~/react/masonry";
+import Card from "~/react/components/Card";
+import CardMarkdownContent from "~/react/components/CardMarkdownContent";
+import { useAppMount } from "~/react/context/app-mount-provider";
+import { getResourcePath } from "~/react/utils/render-utils";
+
+export default function Gallery({ files }: { files: TFile[] }) {
+  const { app } = useAppMount();
+  const breakpointColumnsObj = {
+    default: 10,
+    3100: 9,
+    2700: 8,
+    2300: 7,
+    1900: 6,
+    1500: 5,
+    1000: 4,
+    700: 3,
+    400: 2,
+  };
+
+  const showItems = (files: TFile[]) => {
+    const items = [];
+    for (let i = 0; i < files.length; i++) {
+      const file = files[i];
+      if (file && file.extension === "md") {
+        items.push(
+          <Card key={file.name} file={file}>
+            <CardMarkdownContent file={file} />
+          </Card>,
+        );
+      } else if (file) {
+        items.push(
+          <Card key={file.name} file={file}>
+            <img src={getResourcePath(app, file.path)} />
+          </Card>,
+        );
+      }
+    }
+    return items;
```

Comment:
[♻️](https://github.com/erikthedeveloper/code-review-emoji-guide#:~:text=:recycle:) This whole part can be rewritten as a simple `files.map(...)`

Reply:
this is only a `for (...; i < files.length; ...)` temporarily - it should be `for (...; i < numRendered; ...)` where `numRendered` is a `useState` which get's updated when you reach the bottom of the infinite scroll and new files are inserted...

I don't think it's a good idea to have to render 10,000 cards if this folder has 10,000 files... so we should stick to some method of pagination. 

This is also related to the wider caching issue: if I, the user, scroll through 600 cards and render them, and `numRendered` is incremented by 100, I **don't** want to open the gallery document at the same scrolling position (obsidian passed on backward/forward navigation) and only see 100 - I should see the 600 which I scrolled through... does that make sense? any ideas?